src/lib/y2partitioner/dialogs/main.rb

```diff
@@ -40,7 +43,13 @@ def run(system, initial)
         res = nil
         loop do
           res = super()
-          break if res != :redraw
+
+          next if res == :redraw
+          if res == :abort && Yast::Mode.installation
```

Comment:
`devicegraph1 == devicegraph2` should give that information, because `==` compares the structure of the whole devicegraph. But, at least in the case of installation, we need to ask for confirmation anyways, because returning `:abort` implies aborting the installation as a whole (that's the behavior of the "abort" button all along the installation).

So, in fact, the behavior on installed system and on installation are different (take into account I'm talking here about the desired behavior during installation, not the existing one that is inconsistent with the rest of the installation steps):

- In an installed system a confirmation is asked for both `:abort` and `:back` (with slightly different message) but, in both cases, only if the devicegraph changed. Of course, this is subject to change if we decide so.
- In installation (to work as any other installation step or workflow "branch"), `:back` should never ask for confirmation but `:abort`always asks about quiting the installation process. No matter if the devicegraph changed or not.

In this pull request I just wanted to focus on the installation behavior since, as I mentioned, we already know what we want (to be consistent with the rest of the installation). For standalone I believe we need to define more things and taking the old one as a reference may not be 100% good. The old one works as a wizard with a summary screen at the end and other stuff that is not preset in the current one and that I don't know if we plan to bring back.